client/cypress/integration/user-list.spec.ts

```diff
@@ -0,0 +1,147 @@
+
+    // Go through each of the cards that are being shown and get the names
+    page.getUserCards().find('.user-card-name')
+      // We should see these users whose age is 27
+      .should('contain.text', 'Stokes Clayton')
+      .should('contain.text', 'Bolton Monroe')
+      .should('contain.text', 'Merrill Parker')
+      // We shouldn't see these users
+      .should('not.contain.text', 'Connie Stewart')
+      .should('not.contain.text', 'Lynn Ferguson');
+  });
+
+  it('Should change the view', () => {
+    // Choose the view type "List"
+    page.changeView('list');
+
+    // We should not see any cards
+    // There should be list items
+    page.getUserCards().should('not.exist');
+    page.getUserListItems().should('exist');
+
+    // Choose the view type "Card"
+    page.changeView('card');
+
+    // There should be cards
+    // We should not see any list items
+    page.getUserCards().should('exist');
+    page.getUserListItems().should('not.exist');
+  });
+
+  it('Should select a role, switch the view, and check that it returned correct elements', () => {
+    // Filter for role 'viewer');
+    page.selectRole('viewer');
+
+    // Choose the view type "List"
+    page.changeView('list');
+
+    // Some of the users should be listed
+    page.getUserListItems().should('have.lengthOf.above', 0);
+
+    // All of the user list items that show should have the role we are looking for
+    page.getUserListItems().each(e => {
+      cy.wrap(e).find('.user-list-role').should('have.text', ' viewer '); // this seems fragile since the spaces are expected
+    });
+  });
+
+  it('Should click view profile on a user and go to the right URL', () => {
+    page.getUserCards().first().then((card) => {
+      const firstUserName = card.find('.user-card-name').text();
+      const firstUserCompany = card.find('.user-card-company').text();
+
+      // When the view profile button on the first user card is clicked, the URL should have a valid mongo ID
+      page.clickViewProfile(page.getUserCards().first());
+
+      // The URL should contain '/users/' (note the ending slash) and '/users/' should be followed by a mongo ID
+      cy.url()
+        .should('contain', '/users/')
+        .should('match', /.*\/users\/[0-9a-fA-F]{24}$/);
+
+      // On this profile page we were sent to, the name and company should be correct
+      cy.get('.user-card-name').first().should('have.text', firstUserName);
+      cy.get('.user-card-company').first().should('have.text', firstUserCompany);
+    });
+   });
+
+  it('Should click add user and go to the right URL', () => {
+    // Click on the button for adding a new user
+    page.addUserButton().click();
+
+    // The URL contains and ends with '/users/new'
+    cy.url()
+      .should('contain', '/users/new')
+      .should(url => expect(url.endsWith('/users/new')).to.be.true);
+
+    // On the page we were sent to, We should see the right title
+    cy.get('.add-user-title').should('have.text', 'New User');
```

Comment:
I might suggest omitting this part of the test—it seems like a bit of a duplicate of what we have in `add-user.spec.ts`.

```suggestion
```

Reply:
I like having a second check that we are indeed at the right page personally.